viewer/main.cpp

```diff
@@ -0,0 +1,213 @@
+  udError res = udPointCloud_Load(pContext, &pModel, "../../samplefiles/HistogramTest.uds", &header);
+  if (res != udE_Success)
+  {
+    printf("Could not load sample UDS file\n");
+    goto epilogue;
+  }
+
+  if (udRenderTarget_SetTargets(pRenderView, pColorBuffer, 0, pDepthBuffer) != udE_Success)
+  {
+    printf("Could not set render target buffers\n");
+    goto epilogue;
+  }
+
+  instance.pPointCloud = pModel;
+  memcpy(instance.matrix, header.storedMatrix, sizeof(header.storedMatrix));
+
+  options.flags = udRCF_None;
+
+  while (isRunning)
+  {
+    double yaw = 0, pitch = 0, tx = 0, ty = 0, tz = 0;
+    int frameTimeMs = (udGetTimeMs() - lastRenderTime);
+    dt = udMin(frameTimeMs / 1000.f, 1 / 60.f); // Clamp dt at 60fps
+    lastRenderTime = udGetTimeMs();
+
+    SDL_Event event;
+    while (SDL_PollEvent(&event))
+    {
+      ImGui_ImplSDL2_ProcessEvent(&event);
+      if (event.type == SDL_QUIT)
+        isRunning = false;
+
+      if (event.type == SDL_MOUSEWHEEL)
+        moveScale = udClamp(moveScale + event.wheel.y, -100, 100);
+      if (event.type == SDL_KEYDOWN && event.key.keysym.sym == SDLK_w)
+        ty += moveSpeed * dt;
+      if (event.type == SDL_KEYDOWN && event.key.keysym.sym == SDLK_s)
+        ty -= moveSpeed * dt;
+      if (event.type == SDL_KEYDOWN && event.key.keysym.sym == SDLK_a)
+        tx -= moveSpeed * dt;
+      if (event.type == SDL_KEYDOWN && event.key.keysym.sym == SDLK_d)
+        tx += moveSpeed * dt;
+    }
+
+    float moveScaler = (moveScale < 0) ? -1.f / moveScale : moveScale + 1;
+    void *pSdlPixels;
+    udUpdateCamera(camera.a, yaw, pitch, tx * moveScaler, ty * moveScaler, tz * moveScaler);
+    SDL_LockTexture(pSdlTexture,
+      NULL,      // NULL means the *whole texture* here.
+      &pSdlPixels,
+      &g_windowWidth);
+
+    if (udRenderTarget_SetMatrix(pRenderView, udRTM_Camera, camera.a) != udE_Success)
+      printf("Could not set render target matrix\n");
+
+    if (udRenderContext_Render(pRenderer, pRenderView, &instance, 1, &options) != udE_Success)
+      printf("Rendering failed!\n");
+
+    //bind pColorBuffer to SDL_window
+    memcpy(pSdlPixels, pColorBuffer, g_windowWidth * g_windowHeight);
+    SDL_UnlockTexture(pSdlTexture);
+
+    //ImGUI
+    ImGui_ImplSDLRenderer_NewFrame();
+    ImGui_ImplSDL2_NewFrame();
+    ImGui::NewFrame();
+
+    ImGui::SetNextWindowSize(ImVec2(0, 0));
+    if (ImGui::Begin("Samples"))
+    {
+      ImGui::LabelText("Samples", "Samples");
+      if (ImGui::BeginListBox("Samples"))
+      {
+        if (ImGui::Selectable("Basic Sample", selectedItem == 0))
+          selectedItem = 0;
+
+        ImGui::EndListBox();
+      }
+      ImGui::End();
+    }
+
+    ImGui::Render();
+    SDL_RenderClear(pSdlRenderer);
+    SDL_RenderCopy(pSdlRenderer, pSdlTexture, NULL, NULL);
+    ImGui_ImplSDLRenderer_RenderDrawData(ImGui::GetDrawData());
+    SDL_RenderPresent(pSdlRenderer);
+  }
+
+epilogue:
+  // Clean up
+  //ImGuiSDL::Deinitialize();
```

Comment:
Is this supposed to be commented out?

Reply:
I made some very minor modifications to the code from what @jproust had, otherwise leaving it exactly as it was. This particular line doesn't compile as the namespace and function don't exist, however, it does serve as a placeholder for cleaning up both ImGui and SDL.

Reply:
Fixed: Removed commented line